### ISSUE TITLE
Results

### ISSUE THREAD

**Issue by dhruva77** [naoto0804/pytorch-inpainting-with-partial-conv#3]
Can you share some of your training results like 5000 iteration 10000 itertation and so on?

**Comment by naoto0804** [comment]
I'm trying to replicate [this](https://github.com/SeitaroShinagawa/chainer-partial_convolution_image_inpainting) implementation, but currently the result is quite worse to show..
I would appreciate it if you have any suggestions

**Comment by Luonic** [comment]
I am already implemented this myself on Tensorflow around month ago, but my results **have not enough clarity and bad structure** in inpainted holes.
I took a look at your losses and I do not see any normalization factor k[n] as in paper for style_out and style_comp losses. But in paper there is I think a mistake in equation if k[n] (page 7):
```
...and Kn is the normalization factor 1 / Cn x Hn x Kn for the n-th selected layer.
```
But we can not use Kn to calculate Kn. 
Looks like it should be "1 / Cn x Hn x Wn" as in [this implementation](https://github.com/SeitaroShinagawa/chainer-partial_convolution_image_inpainting):
```
if i==0: 
            L_style_out = F.mean_absolute_error(hout_gram,hgt_gram)/(C*H*W)
            L_style_comp = F.mean_absolute_error(hcomp_gram,hgt_gram)/(C*H*W)
        else:
            L_style_out += F.mean_absolute_error(hout_gram,hgt_gram)/(C*H*W)
            L_style_comp += F.mean_absolute_error(hcomp_gram,hgt_gram)/(C*H*W)   
```
Maybe this is source of your problem?

**Comment by naoto0804** [comment]
@Luonic
Thank you for the suggestion!
Unfortunately, I implement the normalization of the gram matrix [here](https://github.com/naoto0804/pytorch-inpainting-with-partial-conv/blob/master/loss.py#L10) .

**Comment by XiSHEN0220** [comment]
@naoto0804 

Hi, 
I am also interested in this work.
Have you reproduced the results? 
I cannot get good results as well.

**Comment by naoto0804** [comment]
@XiSHEN0220 Unfortunately no :(

**Comment by SunnerLi** [comment]
@naoto0804 hello～
I found this implementation few days ago.
On the other hand, I also try to re-implement the idea of partial convolution two months ago.
(Not reproduce the paper result)
However, the result I got is also awful...
My project is [here](https://github.com/SunnerLi/P-Conv)
I try to train again recently. The whole training might cause three months.
Hope we can get the better results in the future :-)

**Comment by yu45020** [comment]
@SunnerLi
I guess such model takes a long time to train. But I am wondering whether the authors are over-fitting their data. Some image inpainting are unreasonable good given  large holes. 

I have a question on your project's result: do you down sample the whole image and do inpainting? I guess the output would be different if you only crop a 224x224 patch.  By the way, what learning rate are your using?  I am curious about your training result plot. Here is my training loss and learning rate:
![my](https://user-images.githubusercontent.com/28139045/43182233-6d57138c-8f95-11e8-9bb7-05d626a8e8c7.JPG). 
I am using a batch size of 9 and 512x512 images. 
My project is [here](https://github.com/yu45020/Text_Segmentation_Image_Inpainting)

I open an [issue](https://github.com/naoto0804/pytorch-inpainting-with-partial-conv/issues/6#issue-344299146) on some improvements on partial convolution. I also benchmark your code (Thank you for sharing your implementation).   By using PyTorch's ```masked_fill_```, my code runs faster and more memory efficient.

**Comment by naoto0804** [comment]
I show some results after the training of 200,000 iterations with a learning rate of 0.0002 in [the main page](https://github.com/naoto0804/pytorch-inpainting-with-partial-conv/issues/3). (thanks to the suggestions by #6)
I hope it's helpful @SunnerLi

**Comment by SunnerLi** [comment]
@yu45020 I didn't test for complete training setting as paper does.
The learning rate is concrete without changing dynamically.
On the other hand, thanks for your recommend.
I will try to revise as the faster version if I'm free.
@naoto0804 The result looks great!
Also thanks for everyone's discussion to make the result more and more better!
